File: task-nest-server/models/todo.go
package models

type Todo struct {
	ID       int    `json:"id"`
	Username string `json:"user_name"`
	Task     string `json:"task"`
	Done     bool   `json:"done"`
}

type FoundTodo struct {
	ID   int    `json:"id"`
	Task string `json:"task"`
	Done bool   `json:"done"`
}

type UserTodos struct {
	UserId string      `json:"user_name"`
	Todos  []FoundTodo `json:"todos"`
}

File: task_nest/lib/model/todo.dart
class ToDo {
  int id;
  String todoText;
  bool isDone;

  ToDo({
    required this.id,
    required this.todoText,
    this.isDone = false,
  });

  factory ToDo.fromJson(Map<String, dynamic> json) {
    return ToDo(
      id: json['id'],
      todoText: json['task'],
      isDone: json['done'],
    );
  }

  static List<ToDo> todoList() {
    return [
      ToDo(id: 0, todoText: 'Live',),
    ];
  }
}